addons/mail/models/mail_channel.py

```diff
@@ -337,13 +339,16 @@ def message_receive_bounce(self, email, partner, mail_id=None):
 
     @api.multi
     def _notify_email_recipients(self, message, recipient_ids):
+        # Excluded Blacklisted
+        recipients = self.env['res.partner'].sudo().browse(recipient_ids).read(['id', 'name', 'email', 'is_blacklisted'])
```

Comment:
You don't have to read records. Looping on a record set is sufficient. You do a lot of search/ browse + read; they are not necessary unless you specifically want to reduce the memory footprint or have some specific behavior. Here I don't think it is necessary (I let you check other occurrences of unnecessary read).